### PR TITLE
Convert matrix-utils to typescript

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

File: src/matrix-utils.ts [hunk 1]
```diff
@@ -1,34 +1,49 @@
...
   });
 }
 
-export async function initClient(clientOptions) {
+export async function initClient(
+  clientOptions: ICreateClientOpts
+): Promise<MatrixClient> {
   // TODO: https://gitlab.matrix.org/matrix-org/olm/-/issues/10
   window.OLM_OPTIONS = {};
   await Olm.init({ locateFile: () => olmWasmPath });
```
Comment (robintown) [inline src/matrix-utils.ts:49]: GitHub won't let me comment on the `let indexedDB;` two lines below this, but can it be typed?

File: src/matrix-utils.ts [hunk 2]
```diff
@@ -161,16 +180,16 @@ export async function createRoom(client, name, isPtt = false) {
   console.log({ isPtt });
```
Comment (robintown) [inline src/matrix-utils.ts:180]: While you're here, this should probably be an actual log message rather than a raw object

File: src/matrix-utils.ts [hunk 3]
```diff
@@ -161,16 +180,16 @@ export async function createRoom(client, name, isPtt = false) {
...
   await client.createGroupCall(
-    room_id,
+    createRoomResult.room_id,
     isPtt ? GroupCallType.Voice : GroupCallType.Video,
     isPtt,
     GroupCallIntent.Prompt
   );
 
-  return room_alias || room_id;
+  return createRoomResult.room_id;
```
Comment (robintown) [inline src/matrix-utils.ts:189]: What's the reason for this change? I think this means you'll now no longer get the room alias in the URL when creating a room

Comment (dbkr) [reply src/matrix-utils.ts]: Good catch - I took this out as it wasn't in the types (it's unspecced) but yeah, it does need the room alias. I'm not sure why it fell back to the ID though, so have removed that.

**Comment by dbkr** [comment]
There are quite a few more changes here now, sorry!

**Review by robintown — APPROVED** [review]
